posthog/models/activity_logging/activity_log.py

```diff
@@ -237,54 +239,57 @@ def changes_between(
-                right = _read_through_relation(right)
-
-            if field == "tagged_items":
-                field = "tags"  # or the UI needs to be coupled to this internal backend naming
-
-            if field == "dashboards" and "dashboard_tiles" in filtered_fields:
-                # only process dashboard_tiles when it is present. It supersedes dashboards
-                continue
-
-            if model_type == "Insight" and field == "dashboard_tiles":
-                # the api exposes this as dashboards and that's what the activity describers expect
-                field = "dashboards"
-
-            if left is None and right is not None:
-                changes.append(Change(type=model_type, field=field, action="created", after=right))
-            elif right is None and left is not None:
-                changes.append(Change(type=model_type, field=field, action="deleted", before=left))
-            elif left != right:
-                changes.append(
-                    Change(
-                        type=model_type,
-                        field=field,
-                        action="changed",
-                        before=left,
-                        after=right,
+            try:
+                left = getattr(previous, field, None)
+                if isinstance(left, models.Manager):
+                    left = _read_through_relation(left)
+
+                right = getattr(current, field, None)
+                if isinstance(right, models.Manager):
+                    right = _read_through_relation(right)
+
+                if field == "tagged_items":
+                    field = "tags"
+
+                if field == "dashboards" and "dashboard_tiles" in filtered_fields:
+                    continue
+
+                if model_type == "Insight" and field == "dashboard_tiles":
+                    field = "dashboards"
+
+                # Skip changes related to feature flags
+                if isinstance(left, FeatureFlag) or isinstance(right, FeatureFlag):
+                    continue
+
+                if left is None and right is not None:
+                    changes.append(Change(type=model_type, field=field, action="created", after=right))
+                elif right is None and left is not None:
+                    changes.append(Change(type=model_type, field=field, action="deleted", before=left))
+                elif left != right:
+                    changes.append(
+                        Change(
+                            type=model_type,
+                            field=field,
+                            action="changed",
+                            before=left,
+                            after=right,
+                        )
                     )
-                )
+            except:
+                # If we have any exceptions parsing stuff, we should just skip it
+                # rather than failing the request.  Right?
+                continue
```

Comment:
well previously we'd know about it, but now it would silently fail

we should at least be capturing the exception to sentry here

Reply:
great callout.  I'm actually going to revisit this tomorrow because I want to make this handling better.

In short, we have a tricky situation here where when we delete the targeting flag from a survey, we delete the underlying feature flag, which means that `changes_between` fails because it can't find a relationship that's been deleted.  I need to figure out a way to capture the previous targeting flag state without just relying on the `_read_through_relations` method.  I'll finalize that tomorrow.